examples/main.go

```diff
@@ -5,32 +5,52 @@ import (
 	"fmt"
 	"log"
 
+	"sync"
+
 	"github.com/nuveo/gofn"
+	"github.com/nuveo/gofn/iaas/digitalocean"
 	"github.com/nuveo/gofn/provision"
 )
 
 func main() {
-
+	wait := &sync.WaitGroup{}
 	contextDir := flag.String("contextDir", "./", "a string")
 	dockerfile := flag.String("dockerfile", "Dockerfile", "a string")
 	imageName := flag.String("imageName", "", "a string")
 	remoteBuildURI := flag.String("remoteBuildURI", "", "a string")
 	volumeSource := flag.String("volumeSource", "", "a string")
 	volumeDestination := flag.String("volumeDestination", "", "a string")
 	flag.Parse()
+	wait.Add(1)
+	run(*contextDir, *dockerfile, *imageName, *remoteBuildURI, *volumeSource, *volumeDestination, wait)
+	wait.Add(1)
+	run(*contextDir, *dockerfile, *imageName, *remoteBuildURI, *volumeSource, *volumeDestination, wait)
+	wait.Add(1)
+	run(*contextDir, *dockerfile, *imageName, *remoteBuildURI, *volumeSource, *volumeDestination, wait)
+	wait.Add(1)
+	run(*contextDir, *dockerfile, *imageName, *remoteBuildURI, *volumeSource, *volumeDestination, wait)
+	wait.Add(1)
+	run(*contextDir, *dockerfile, *imageName, *remoteBuildURI, *volumeSource, *volumeDestination, wait)
+	wait.Wait()
 
-	stdout, err := gofn.Run(&provision.BuildOptions{
-		ContextDir: *contextDir,
-		Dockerfile: *dockerfile,
-		ImageName:  *imageName,
-		RemoteURI:  *remoteBuildURI,
-	}, &provision.VolumeOptions{
-		Source:      *volumeSource,
-		Destination: *volumeDestination,
-	})
-	if err != nil {
-		log.Println(err)
-	}
+}
+func run(contextDir, dockerfile, imageName, remoteBuildURI, volumeSource, volumeDestination string, wait *sync.WaitGroup) {
+	go func() {
+		stdout, err := gofn.Run(&provision.BuildOptions{
+			ContextDir: contextDir,
+			Dockerfile: dockerfile,
+			ImageName:  imageName,
+			RemoteURI:  remoteBuildURI,
+			Iaas:       &digitalocean.Digitalocean{},
+		}, &provision.VolumeOptions{
+			Source:      volumeSource,
+			Destination: volumeDestination,
+		})
+		if err != nil {
+			log.Println(err)
+		}
 
-	fmt.Println(stdout)
+		fmt.Println(stdout)
+		wait.Done()
```

Comment:
I would move wait.Done() to the top of the anonymous function and add a defer before it.

Use defer before every critical calls, like sync related calls, at the beginning since you don't know where the code could break. Let's say you don't defer before wg.Done(), and a panic occurs in your anonymous goroutine in your example. If you don't have panic recover, it will panic. But what happens if you have a panic recover? Everything's fine now? No. You will get deadlock at wg.Wait() since your wg.Done() gets skipped because of panic! However, by using defer, this wg.Done() will be executed at the end, even if panic happened. 

I just think it is good practice to use a defer in these cases.